api/src/domain-logic/analytics.js

```diff
@@ -0,0 +1,61 @@
+const dayjs = require('dayjs')
+
+const supabase = require('./supabase')
+
+const today = () => dayjs().format('YYYY-MM-DD')
+const TABLE =
+  process.env.NODE_ENV === 'production' ? 'analytics' : 'analytics_dev'
+
+/* Note
+ * - Cannot use upsert because in order to make upsert correct, should use id instead
+ */
+
+const updateAnalytic = ({ count, ...item }) =>
+  supabase
+    .from(TABLE)
+    .update(
+      { ...item, count: count + 1 },
+      { returning: 'minimal', count: 'exact' },
+    )
+    .match({ id: item.id })
+
+const fetchAnalytic = ({ meta, action }) =>
+  supabase
+    .from(TABLE)
+    .select('*', { head: false, count: 'exact' })
+    .eq('date', today())
+    .eq('meta->>word', meta?.word)
```

Comment:
`meta?.word` could be `undefined`. Are we sure we want to update when that happened? And "undefined" is a valid English word.